js/logging.js

```diff
@@ -49,6 +49,10 @@ loggingModule.factory(
                 $log.error.apply($log, arguments);
             }
 
+            if($window.talisLogger.onBeforeLogError && !$window.talisLogger.onBeforeLogError(exception)) {
+                return false;
+            }
+
```

Comment:
I think this name makes sense, although I wonder if `shouldLogError` would be clearer?

Reply:
Another thing to note on this: my original plan was to pass the callback in through `LOGGING_CONFIG` but we can only support json in that and not functions, we do some similar things to this with `$rootScope` but since we're trying to avoid adding to that I went with `$window` as that seemed like the next best place to pass this information across.

Reply:
`shouldLogError` seems clearer but an example of it being used by a consumer of this service would help determine that for sure.

Reply:
I've got the corresponding Elevate PR ready here: https://github.com/talis/elevate-app/pull/3263/files where I'm just hooking this straight up to an error filtering service, in that the function is called `shouldLogError` as that's the goal of that function, to indicate whether a particular error should be logged. But from the perspective of the logger I'm less sure, it feels more like a pre action hook, but one that can change the behaviour which is why I'm in two minds on the name.

Reply:
It is a pre-action hook but here, it is still explicitly preventing the logging for occuring if the provided function returns false.  How about `shouldFilterLogFunction`?:

```javascript
$window.talisLogger.shouldFilterLogFunction = errorFilterService.shouldLogError;
```